### PR TITLE
Fix multiple clients created for same route not using respective config

### PR DESCRIPTION
**Pull request by chamil321**
## Purpose
Currently the Client connection pool key is scheme + host + port. Regardless what the client config properties are, Clients created for same route share one pool. This fix is to differentiate the pools creation and consider configs. 
With this change pools are created per client. 

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1727
Related issue : https://github.com/ballerina-platform/ballerina-standard-library/issues/1811

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina-tests/tests/http2_mutual_ssl_test.bal [hunk 1]
```diff
@@ -123,7 +123,7 @@ http:ClientConfiguration http2MutualSslClientConf2 = {
     http2Settings: { http2PriorKnowledge: true }
 };
 
-@test:Config {}
+//@test:Config {}
```
Comment (chamil321) [inline ballerina-tests/tests/http2_mutual_ssl_test.bal:126]: Disabled temporarily until check this with bashinee. Test needs to be changed.

**Comment by shafreenAnfar** [comment]
The solution looks good to me. But there are a bunch of disabled test cases to be sorted before merging right ?

**Comment by shafreenAnfar** [comment]
Also any reason why we haven't updated the change log ?

**Comment by chamil321** [comment]
> The solution looks good to me. But there are a bunch of disabled test cases to be sorted before merging right ?

Those disable test cases will be enabled via https://github.com/ballerina-platform/module-ballerina-http/pull/605.
Updated changelog

**Review by ldclakmal — APPROVED** [review]

**Pull request merged**